### ISSUE TITLE
[NextJS] [ graphql-let ] graphql-codegen error

### ISSUE THREAD

**Issue by ominatu1** [Sitecore/jss#1282]
## Description
graphql-codegen error after update GraphQL Introspection:

Step to reproduce:
dotnet sitecore login --authority https://demo.identityserver.local --cm https://demo.cm.local --allow-write true
jss graphql:update
jss build

## Expected behavior
"jss build" should successful but have error relate to graphql-codegen

[ graphql-let ] Running graphql-codegen...
[ graphql-let ] Interface field Item.children expects type ItemSearchResults! but App.children is type ItemSearchResults.
Interface field Item.children expects type ItemSearchResults! but C__Route.children is type ItemSearchResults.
Interface field Item.children expects type ItemSearchResults! but C__StandardTemplate.children is type ItemSearchResults.

## Environment
* Sitecore Version: 10 (same issue in 10.2, 10.3)
![image](https://user-images.githubusercontent.com/6355912/210994240-f72d0c82-34ae-471c-82d6-b9494abc8c2a.png)

![image](https://user-images.githubusercontent.com/6355912/210993363-e7c0a18b-a9d2-40fe-a237-4bab68691257.png)

**Comment by larserhardsen** [comment]
We see the same behavior, and it seems to be related to Headless Services 21 - version 20 does not appear to have this problem.

**Comment by ambrauer** [comment]
@donggach @larserhardsen Thanks for the information, I've added this to our backlog to investigate / fix in a future release.

**Comment by bewards** [comment]
What is the preferred workaround for this issue with SC 10.3 Headless 21? I had to disable **all** graphql files, which means graphql connected mode can't be used. Then I had to update the entrypoint for docker rendering and the nextjs app to a new start:connected command so that when it fails to find graphql files for graphql-let, it still passes through and let's the rendering app initialize:

```json
"start:connected_continue-on-error": "npm run bootstrap & npm-run-all --parallel next:dev start:watch-components"
```

**Comment by ambrauer** [comment]
A hotfix for this issue is now available, upon request, for Sitecore XP 10.3 and Headless Services 21.0.0. Please reach out to Sitecore Support and they will be happy to assist. You can use reference number 572005 in your request.

If you are using Sitecore XM Cloud, this issue should be fixed in the latest version.

cc @ominatu1 @larserhardsen @bewards
